### PR TITLE
Refresh FFM API documents

### PR DESCRIPTION
**Pull request by mcimadamore**
This patch refreshes the FFM API documents to reflect the latest API changes. There has been some reshuffling: I realized that we had two sections on unsafe memory segments, one at the end of the memory document, another at the beginning of the FFI document. Since unsafe segments are more common when interacting with native functions, I just kept the latter (and enhanced to explain the options in more details).

I also moved (and rewrote) the segment allocator section from the FFI doc to the memory document, as I think working with allocators and using custom arenas is more generally useful, even outside FFI support.

I also cleaned up a lot of minor incongruences in the code snippets, text and footnotes which still referred to old API names and concepts.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Reviewers
 * [Jorn Vernee](https://openjdk.org/census#jvernee) (@JornVernee - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/panama-foreign pull/805/head:pull/805` \
`$ git checkout pull/805`

Update a local copy of the PR: \
`$ git checkout pull/805` \
`$ git pull https://git.openjdk.org/panama-foreign pull/805/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 805`

View PR using the GUI difftool: \
`$ git pr show -t 805`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/panama-foreign/pull/805.diff">https://git.openjdk.org/panama-foreign/pull/805.diff</a>

</details>

### CODE REVIEW

**Comment by mcimadamore** [comment]
Documents can be viewed here:
https://github.com/mcimadamore/panama-foreign/blob/foreign_doc_refresh/doc/panama_memaccess.md
https://github.com/mcimadamore/panama-foreign/blob/foreign_doc_refresh/doc/panama_ffi.md

File: doc/panama_ffi.md [hunk 1]
```diff
@@ -12,75 +12,35 @@ Before we dive into the specifics of the foreign function support, it would be u
...
-```java
-try (Arena arena = Arena openConfined()) {
-    MemorySegment arr = MemorySegment.allocateNative(MemoryLayout.sequenceLayout(5, JAVA_INT), arena.scope());
-    for (int i = 0 ; i < 5 ; i++) {
-        arr.setAtIndex(JAVA_INT, i, i);
-    }
+MemorySegment foreign = null;
+try (Arena arena = Arena.ofConfined()) {
+      foreign = someSegment.get(ValueLayout.ADDRESS, 0)           // size = 0, scope = always alive
+                           .reinterpret(4, arena.scope(), null);  // size = 4, scope = arena.scope()
```
Comment (JornVernee) [inline doc/panama_ffi.md:29]: This should pass the arena now, I think.
```suggestion
                           .reinterpret(4, arena, null);  // size = 4, scope = arena.scope()
```

**Review by JornVernee — APPROVED** [review]

File: doc/panama_ffi.md (hunk 1 above)
Comment (mcimadamore) [reply doc/panama_ffi.md:29]: Good catch - I'll re-check. Main issue is that the edits were done when the method accepted a scope. Perhaps some other editing is required.

**Comment by mcimadamore** [comment]
I've addressed the comments and also fixed some other issues:
* minor typos
* double whitespaces
* missing `java` in some code blocks

**Review by JornVernee — APPROVED** [review]

**Pull request closed without merging**